### PR TITLE
SYCL: Print submission command queue property

### PR DESCRIPTION
**Pull request by masterleinad**
Related to #6912 and #7003. Prior to https://github.com/intel/llvm/pull/12279 and https://github.com/oneapi-src/unified-runtime/pull/1218, the oneAPI implementation didn't support immediate command lists. 
There are different ways to make sure to use the standard command queue:
- constructing queues with the `sycl::ext::intel::property::queue::no_immediate_command_list` property (see https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_queue_immediate_command_list.asciidoc)
- setting the `SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS` environment variable.

This pull request just detects if given queues have been constructed with either property. This should give enough information to diagnose potential problems with Graphs. The intended way to fix any related issues is to set `SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS`.

### CODE REVIEW

File: core/src/SYCL/Kokkos_SYCL.cpp [hunk 1]
```diff
@@ -110,6 +110,19 @@ void SYCL::print_configuration(std::ostream& os, bool verbose) const {
...
+              sycl::ext::intel::property::queue::immediate_command_list>())
+    os << "Immediate command lists enforced\n";
+  else if (sycl_queue()
+               .has_property<sycl::ext::intel::property::queue::
+                                 no_immediate_command_list>())
+    os << "Standard command queue enforced\n";
+  else
+#endif
+    os << "Immediate command lists and standard command queue allowed. "
+          "SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS takes precedence.\n";
```
Comment (dalg24) [inline core/src/SYCL/Kokkos_SYCL.cpp:125]: Is that an environment variable?  If so you might want to do `std::getenv()` and spell out if it is defined and wha the value is.

File: core/src/SYCL/Kokkos_SYCL.cpp [hunk 2]
```diff
@@ -110,6 +110,25 @@ void SYCL::print_configuration(std::ostream& os, bool verbose) const {
...
+    os << "Standard command queue enforced\n";
+  else
+#endif
+  {
+    os << "Immediate command lists and standard command queue allowed. "
+          "SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS takes precedence.\n";
+    if (const char* environment_setting =
+            std::getenv("SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS"))
+      os << "SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS: "
+         << environment_setting << '\n';
```
Comment (dalg24) [inline core/src/SYCL/Kokkos_SYCL.cpp:130]:
```suggestion
    os << "Immediate command lists and standard command queue allowed.\n"
        if (const char* environment_setting =
            std::getenv("SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS")) 
         os<< "SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=" << environment_setting <<  " takes precedence.\n";
         else
      os << "SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS not defined.\n";
```

Comment (dalg24) [reply core/src/SYCL/Kokkos_SYCL.cpp:130]: Obviously not formatted.  If you edit, I would prefer the `ENV=VAL` format over `ENV: VAL`.

**Comment by masterleinad** [comment]
Only `CUDA-11.6-NVCC-DEBUG` failed with connectivity issues on `fetnat03`.

**Review by nliber — APPROVED** [review]

**Comment by masterleinad** [comment]
> @masterleinad does this merit a 4.4 changelog entry?

Added.

**Pull request merged**